bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/IFindReplaceLogic.java

```diff
@@ -0,0 +1,148 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Vector Informatik GmbH and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Vector Informatik GmbH - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.ui.internal.findandreplace;
+
+import org.eclipse.swt.widgets.Display;
+
+import org.eclipse.jface.text.IFindReplaceTarget;
+
+/**
+ * Implements a generalized logic operator for in-file Find/Replace-Operations.
+ * Requires a target that inherits from {@link IFindReplaceTarget} to operate.
+ * Allows enabling or disabling different {@link SearchOptions} which will be
+ * applied to subsequent operations.
+ */
+public interface IFindReplaceLogic {
+
+	/**
+	 * Activate a search option
+	 *
+	 * @param searchOption option
+	 */
+	public void activate(SearchOptions searchOption);
+
+	/**
+	 * Deactivate a search option
+	 *
+	 * @param searchOption option
+	 */
+	public void deactivate(SearchOptions searchOption);
+
+	/**
+	 * @param searchOption option
+	 * @return whether the option is active
+	 */
+	public boolean isActive(SearchOptions searchOption);
+
+	/**
+	 * Returns the current status of FindReplaceLogic. The Status can inform about
+	 * events such as an error happening, a warning happening (e.g.: the
+	 * search-string wasn't found) and brings a method to retrieve a message that
+	 * can directly be displayed to the user.
+	 *
+	 * @return FindAndReplaceMessageStatus
+	 */
+	IFindReplaceLogicStatus getStatus();
+
+	boolean isRegExSearchAvailableAndActive();
+
+	/**
+	 * Tells the dialog to perform searches only in the scope given by the actually
+	 * selected lines.
+	 *
+	 * @param selectedLines <code>true</code> if selected lines should be used
+	 */
+	void useSelectedLines(boolean selectedLines);
```

Comment:
Isn't this a `SearchOption` as well?

Reply:
Yes it is, and this method is even really confusing since we only want to refer to "Global vs non-global" search. I have removed this method